app/src/common/chromium/com/igalia/wolvic/browser/api/impl/TabWebContentsDelegate.java

```diff
@@ -0,0 +1,58 @@
+package com.igalia.wolvic.browser.api.impl;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.igalia.wolvic.browser.api.WSession;
+
+import org.chromium.blink.mojom.DisplayMode;
+import org.chromium.components.embedder_support.delegate.WebContentsDelegateAndroid;
+import org.chromium.content_public.browser.WebContents;
+
+public class TabWebContentsDelegate extends WebContentsDelegateAndroid {
+    private @NonNull SessionImpl mSession;
+
+    private boolean mIsFullscreen;
+
+    public TabWebContentsDelegate(WebContents webContents, @NonNull SessionImpl session) {
+        mSession = session;
+        mSession.getTab().setWebContentsDelegate(webContents, this);
+    }
+
+   @Override
+   public boolean takeFocus(boolean reverse) {
+       @Nullable WSession.ContentDelegate delegate = mSession.getContentDelegate();
+       if (delegate != null && !reverse) {
+           delegate.onFocusRequest(mSession);
+       }
+       return false;
+   }
+
+    @Override
+    public void enterFullscreenModeForTab(boolean prefersNavigationBar, boolean prefersStatusBar) {
+        @Nullable WSession.ContentDelegate delegate = mSession.getContentDelegate();
+        if (delegate != null) {
```

Comment:
In general we prefer early returns to avoid extra indentation as you perfectly did in most of the cases.. So in this case
```
if (delegate==null)
 return;
```

Reply:
Done